File: scripts/generate_postal_codes.mjs
import { mkdirSync, writeFileSync } from "node:fs";
import { dirname, resolve } from "node:path";

function generateCodes(prefix) {
  const start = prefix * 1000;
  const end = start + 1000;
  const codes = [];
  for (let code = start; code < end; code += 1) {
    codes.push(String(code).padStart(5, "0"));
  }
  return codes;
}

const codes = [...generateCodes(37), ...generateCodes(49)].sort();
const output = resolve(process.cwd(), "data", "postal_codes.json");
mkdirSync(dirname(output), { recursive: true });
writeFileSync(output, JSON.stringify(codes, null, 2), "ascii");

File: scripts/generate_postal_codes.py
import json
from pathlib import Path


def generate_codes(prefix: int) -> list[str]:
    start = prefix * 1000
    end = start + 1000
    return [f"{code:05d}" for code in range(start, end)]


def main() -> None:
    codes = sorted(generate_codes(37) + generate_codes(49))
    output = Path(__file__).resolve().parents[1] / "data" / "postal_codes.json"
    output.parent.mkdir(parents=True, exist_ok=True)
    output.write_text(json.dumps(codes, indent=2), encoding="ascii")


if __name__ == "__main__":
    main()
